scipy/_lib/_util.py

```diff
@@ -709,22 +709,29 @@ def _nan_allsame(a, axis, keepdims=False):
 
 def _contains_nan(a, nan_policy='propagate', use_summation=True,
                   policies=None):
-    if not isinstance(a, np.ndarray):
+    # _contains_nan expects `xp` to be an array-api compatible namespace
+    # (e.g. from scipy._lib.array_api_compat).
+    xp = array_namespace(a)
```

Comment:
Can you follow the pattern of allowing the option to pass `xp` around private functions as a kwarg to avoid duplicate `array_namespace` calls:

```python
def _contains_nan(a, nan_policy='propagate', use_summation=True,
                  policies=None, *, xp=None):
	if xp is None:
		xp = array_namespace(a)
```

Reply:
Probably. This was done in pieces with the lowest level functions first, so the higher-level functions weren't passing `xp` yet.

Reply:
Done.